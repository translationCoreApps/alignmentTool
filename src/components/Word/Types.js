export const SECONDARY_WORD = 'secondary';
export const PRIMARY_WORD = 'primary';
