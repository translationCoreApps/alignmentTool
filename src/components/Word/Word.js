import React, { Component } from 'react';
import PropTypes from 'prop-types';
import WordOccurrence from './WordOccurrence';

const defaultStyles = {
  borderLeft: '5px solid #44C6FF',
  padding: '5px',
  backgroundColor: '#FFFFFF',
  boxShadow: "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
  cursor: 'move',
  display: 'flex',
  flexDirection: 'row'
};

/**
 * Renders a single word
 * @property {string} word - the represented word
 * @property {int} occurrence
 * @property {int} occurrences
 */
class Word extends Component {
  render() {
    const { word, occurrence, occurrences, style } = this.props;
    const wordStyle = {...defaultStyles, ...style};
    return (
      <div style={wordStyle}>
        <span style={{flex: 1}}>
          {word}
        </span>
        <WordOccurrence occurrence={occurrence}
                        occurrences={occurrences}/>
      </div>
    );
  }
}

Word.propTypes = {
  style: PropTypes.object,
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired
};

Word.defaultProps = {
  style: {}
};

export default Word;
