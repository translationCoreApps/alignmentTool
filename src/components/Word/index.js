import React from 'react';
import PropTypes from 'prop-types';
import Word from './Word';
import { DragSource } from 'react-dnd';
import ItemTypes from '../ItemTypes';

/**
 * Renders a draggable word.
 *
 * @property {string} word - the represented word
 * @property {int} occurrence
 * @property {int} occurrences
 */
class DraggableWord extends React.Component {
  render() {
    const {connectDragSource, word, occurrence, occurrences, isDragging} = this.props;
    const opacity = isDragging ? 0.4 : 1;

    return connectDragSource(
      <div>
        <Word word={word}
              style={{opacity}}
              occurrence={occurrence}
              occurrences={occurrences}/>
      </div>
    );
  }
}

DraggableWord.propTypes = {
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired,
  connectDragSource: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired,
  alignmentIndex: PropTypes.number
};

DraggableWord.defaultProps = {
  alignmentIndex: undefined // just to be explicit
};

/**
 * Handles drag events on the word
 */
const dragHandler = {
  beginDrag(props) {
    // Return the data describing the dragged item
    return {
      word: props.word,
      occurrence: props.occurrence,
      occurrences: props.occurrences,
      alignmentIndex: props.alignmentIndex,
      type: ItemTypes.SECONDARY_WORD
    };
  }
};

export default DragSource(
  ItemTypes.SECONDARY_WORD,
  dragHandler,
  (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  })
)(DraggableWord);
