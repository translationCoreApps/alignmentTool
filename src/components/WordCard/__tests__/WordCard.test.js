import React from 'react';
import renderer from 'react-test-renderer';
import Word from '../index';

it('occurs once', () => {
  const wrapper = renderer.create(
    <Word occurrences={1}
          occurrence={1}
          word={'hello'}/>
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders rtl', () => {
  const wrapper = renderer.create(
    <Word occurrences={1}
          occurrence={1}
          direction={'rtl'}
          word={'hello'}/>
  );
  expect(wrapper).toMatchSnapshot();
});

it('occurs multiple times', () => {
  const wrapper = renderer.create(
    <Word occurrences={2}
          occurrence={1}
          word={'hello'}/>
  );
  expect(wrapper).toMatchSnapshot();
});

it('is a suggestion', () => {
  const wrapper = renderer.create(
    <Word occurrences={2}
          occurrence={1}
          isSuggestion={true}
          word={'hello'}/>
  );
  expect(wrapper).toMatchSnapshot();
});
