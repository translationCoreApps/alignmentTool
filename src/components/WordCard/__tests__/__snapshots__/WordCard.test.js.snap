// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is a suggestion 1`] = `
<div
  aria-describedby={null}
  className=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  title={null}
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderLeft": "5px solid #1b7729",
        "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "row",
        "padding": "9px",
      }
    }
  >
    <span
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <span
        className={undefined}
        onClick={[Function]}
        style={
          Object {
            "flexGrow": 2,
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "width": "max-content",
          }
        }
      >
        hello
      </span>
      <svg
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#646464",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 20,
            "marginLeft": 5,
            "muiPrepared": true,
            "opacity": 0.5,
            "transition": "0.1s",
            "userSelect": "none",
            "verticalAlign": "middle",
            "width": 20,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </span>
    <sup
      style={
        Object {
          "opacity": "0.8",
          "padding": "2px 0px 0px",
          "top": "0px",
        }
      }
    >
      1
    </sup>
  </div>
</div>
`;

exports[`occurs multiple times 1`] = `
<div
  aria-describedby={null}
  className=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  title={null}
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderLeft": "5px solid #44C6FF",
        "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "row",
        "padding": "9px",
      }
    }
  >
    <span
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <span
        className={undefined}
        onClick={[Function]}
        style={
          Object {
            "flexGrow": 2,
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "width": "max-content",
          }
        }
      >
        hello
      </span>
    </span>
    <sup
      style={
        Object {
          "opacity": "0.8",
          "padding": "2px 0px 0px",
          "top": "0px",
        }
      }
    >
      1
    </sup>
  </div>
</div>
`;

exports[`occurs once 1`] = `
<div
  aria-describedby={null}
  className=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  title={null}
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderLeft": "5px solid #44C6FF",
        "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "row",
        "padding": "9px",
      }
    }
  >
    <span
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <span
        className={undefined}
        onClick={[Function]}
        style={
          Object {
            "flexGrow": 2,
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "width": "max-content",
          }
        }
      >
        hello
      </span>
    </span>
  </div>
</div>
`;

exports[`renders rtl 1`] = `
<div
  aria-describedby={null}
  className=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  title={null}
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRight": "5px solid #44C6FF",
        "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "row",
        "padding": "9px",
      }
    }
  >
    <span
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <span
        className={undefined}
        onClick={[Function]}
        style={
          Object {
            "flexGrow": 2,
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "width": "max-content",
          }
        }
      >
        hello
      </span>
    </span>
  </div>
</div>
`;
